tdi_python/tdicli.py

```diff
@@ -0,0 +1,2277 @@
+          "Pipe id: {}\n"
+          "Direction: {}\n"
+          "Parser id: {}\n"
+          "sel_grp_id: {}\n"
+          "act_mbr_id: {}\n"
+          "logical_table_index: {}\n"
+          "is_add: {}\n"
+          .format(dev_id, pipe_id, direction, parser_id, sel_grp_id, act_mbr_id, logical_table_index, is_add))
+
+def _default_operation_callback(dev_id, pipe_id, direction, parser_id):
+    print("operation complete - dev_id: {}, pipe_id: {}, direction: {}, parser_id: {}".format(dev_id, pipe_id, direction, parser_id))
+
+def _learn_fields_print(dev_id, pipe_id, direction, parser_id, session, data):
+    print("Device id: {}\n"
+          "Pipe id: {}\n"
+          "Direction: {}\n"
+          "Parser id: {}\n".format(dev_id, pipe_id, direction, parser_id))
+    print("Learn data:\n{}".format(data))
+    return 0
+
+class TDILeaf(TDIContext):
+    """
+    This class creates easy to type, autocompleted python entrypoints to the
+    TDI Runtime API. Each instance of the class represents one table. It exposes
+    Add, modify commands for each action type (or one of each for the table if
+    said table has no actions). It also exposes get and dump commands for
+    retrieving information from TDI Runtime.
+    """
+    def __init__(self, name, c_tbl, cintf, parent_node=None, children=None):
+        self._set_name(name)
+        self._c_tbl = c_tbl
+        self._cintf = cintf
+        c_tbl.set_frontend(self)
+        self._children = {}
+        #
+        # Link the TDI Command Object Tree
+        #
+        if parent_node is not None and isinstance(parent_node, TDINode):
+            self._parent_node = parent_node
+            self._parent_node._add_child(self)
+        else:
+            self._parent_node = None
+        #
+        # If some children were provided add them
+        #
+        if children is not None:
+            for c in children:
+                self._add_child(c)
+        #
+        # Setup Methods and Attributes for Leaf Command Node
+        #
+        self._adds = {}
+        self._mods = {}
+        self._mods_inc = {}
+        self._dels = []
+        self._gets = []
+        self._entries = {}
+        self._init_table_methods()
+        self._init_string_choices()
+        for cmd in self._c_tbl.supported_commands:
+            try:
+                self._children[cmd] = getattr(self,cmd)
+            except:
+                pass
+                # print("CLI log: Command {} is not ready or available".format(cmd))
+                # pdb.set_trace()
+        self._set_docstring()
+        # if self._c_tbl.table_type_map(self._c_tbl.get_type()) in self._c_tbl.unimplemented_tables:
+        #     print("CLI Err: Unimplemented Command Object {}".format(self._name))
+        #     return
+        # else:
+        #     print("CLI Log: Implemented Command Object {}".format(self._name))
+        #     for cmd in self._children:
+        #         print("CLI log: Command {}".format(cmd))
```

Comment:
Any reason for commenting these out ?

Reply:
This was from bfrtcli.py and I will keep it in case we want enable this capability to check inimplemented_tables.
This not supported feature now: bfrtTable.py:        self.unimplemented_tables = ["TODO"]